api/v1/storagecluster_types.go

```diff
@@ -89,6 +89,11 @@ type StorageClusterSpec struct {
 	// Defaults to false
 	// +optional
 	EnableCephTools bool `json:"enableCephTools,omitempty"`
+
+	// Logging represents loggings settings
+	// +optional
+	// +nullable
+	LogCollector rookCephv1.LogCollectorSpec `json:"logCollector,omitempty"`
```

Comment:
Let's make this nullable so we can detect if we should use defaults or not.
```suggestion
	LogCollector *rookCephv1.LogCollectorSpec `json:"logCollector,omitempty"`
```